eox_tagging/test/test_models.py

```diff
@@ -504,3 +506,118 @@ def test_soft_delete_queryset(self):
         Tag.objects.delete()
 
         self.assertFalse(Tag.objects.active())
+
+
+class TestTagQuerysetManager(TestCase):
+    """
+    Test cases for queryset used as tag manager. This test cases are focused to
+    check how the method behaves with objects like CourseOverview and CourseEnrollment.
+    """
+
+    def setUp(self):
+        self.tagQueryset = TagQuerySet()
+
+    @patch.object(OpaqueKeyProxyModel, 'objects')
+    def test_create_tag_with_course(self, opaque_objects_mock):
+        """Test tagging a course using TagQueryset manager."""
+        self.tagQueryset.create = Mock()
+        course_mock = Mock()
+        opaque_mock = Mock()
+        opaque_objects_mock.get_or_create.return_value = opaque_mock, Mock()
+        course_mock.__class__.__name__ = "CourseOverview"
+        kwargs = {
+            "target_object": course_mock,
+        }
+
+        self.tagQueryset.create_tag(**kwargs)
+
+        self.tagQueryset.create.called_once_with(target_object=opaque_mock)
+
+    # pylint: disable=protected-access
+    def test_find_all_tags_for_course(self):
+        """Test getting tags with course as target."""
+        target_type, target_id = "CourseOverview", 1
+        target, target_ctype = Mock(), Mock()
+        self.tagQueryset._get_object_for_this_type = Mock()
+        self.tagQueryset._get_object_for_this_type.return_value = target, target_ctype
+        self.tagQueryset.filter = Mock()
+        target.values_list.return_value = [1]
+
+        self.tagQueryset.find_all_tags_for(target_type, target_id)
+
+        self.tagQueryset._get_object_for_this_type.assert_called_once_with(
+            "opaquekeyproxymodel",
+            target_id
+        )
+        self.tagQueryset.filter.assert_called_once_with(
+            target_type=target_ctype,
+            target_object_id__in=[1],
+        )
+
+    @patch.object(ContentType, 'objects')
+    def test_get_objects_for_type_user(self, content_type_mock):
+        """Test getting courses associated with tags."""
+        ctype_object = Mock()
+        content_type_mock.get.return_value = ctype_object
```

Comment:
Also you can check that content_type_mock.get.assert_called_once_with(model='User')